python-spec/src/somacore/spatialdata.py

```diff
@@ -0,0 +1,763 @@
+    #     class MultiscaleImage(  # type: ignore[type-var]
+    #         ImplBaseCollection[ImplSOMAObject],
+    #         somacore.MultiscaleImage[ImplDenseNDArray, ImpSOMAObject],
+    #     ):
+    #         ...
+
+    soma_type: Final = "SOMAMultiscaleImage"  # type: ignore[misc]
+    __slots__ = ()
+
+    # Lifecycle
+
+    @classmethod
+    @abc.abstractmethod
+    def create(
+        cls,
+        uri: str,
+        *,
+        type: pa.DataType,
+        image_type: str = "CYX",
+        reference_level_shape: Sequence[int],
+        axis_names: Sequence[str] = ("c", "x", "y"),
+        platform_config: Optional[options.PlatformConfig] = None,
+        context: Optional[Any] = None,
+    ) -> Self:
+        """Creates a new collection of this type at the given URI.
+
+        Args:
+            uri: The URI where the collection will be created.
+            axis_names: The names of the axes of the image.
+            reference_level_shape: The shape of the reference level for the multiscale
+                image. In most cases, this should correspond to the size of the image
+                at ``level=0``.
+            image_type: The order of the image axes using standard names. See
+                :class:`ImageProperties` for more details.
+
+        Returns:
+            The newly created collection, opened for writing.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def add_new_level(
+        self,
+        key: str,
+        *,
+        uri: Optional[str] = None,
+        shape: Sequence[int],
+    ) -> data.DenseNDArray:
+        """Add a new level in the multi-scale image.
+
+        Parameters are as in :meth:`data.DenseNDArray.create`. The provided shape will
+        be used to compute the scale between images and must correspond to the image
+        size for the entire image.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    # Data operations
+
+    @abc.abstractmethod
+    def read_region(
+        self,
+        level: Union[int, str],
+        region: options.SpatialRegion = (),
+        *,
+        channel_coords: options.DenseCoord = None,
+        transform: Optional[coordinates.CoordinateTransform] = None,
+        region_coord_space: Optional[coordinates.CoordinateSpace] = None,
+        create_mask: bool = False,
+        result_order: options.ResultOrderStr = _RO_AUTO,
+        platform_config: Optional[options.PlatformConfig] = None,
+    ) -> "SpatialRead[Union[pa.Tensor, pa.Table]]":
+        """Reads a user-defined region into a :class:`SpatialRead` with data in
+        either an Arrow tensor or table.
+
+        Reads the bounding box of the input region from the requested image level. If
+        ``create_mask=True``, this will return a :class:`SpatialRead` with the image
+        data and mask stored in a :class`pa.Table`. Otherwise, the data will be stored
+        as a :class:`pa.Tensor`.
```

Comment:
What is the format of that table? Also what is the size of the returned image if `create_mask=True`?

Reply:
From the meeting: 
* If supported by arrow, we should return a nullable `pyarrow.Tensor` instead of a `Table`. If that isn't supported, we should return tuple of `pa.Table` and we should add additional docs here about the type/size of the mask.

Reply:
I went ahead and dropped the mask for now.